lib/AST/ASTPrinter.cpp

```diff
@@ -4785,12 +4787,26 @@ void PrintAST::visitErrorExpr(ErrorExpr *expr) {
   Printer << "<error>";
 }
 
-void PrintAST::visitTernaryExpr(TernaryExpr *expr) {}
+void PrintAST::visitTernaryExpr(TernaryExpr *expr) {
+  if (auto condExpr = expr->getCondExpr(); condExpr != nullptr) {
+    if (auto elseExpr = expr->getElseExpr(); elseExpr != nullptr) {
+      visit(expr->getCondExpr());
+      Printer << " ? ";
+      visit(expr->getThenExpr());
+      Printer << " : ";
+      visit(expr->getElseExpr());
+    }
+  }
```

Comment:
Sorry I should have been clearer, we still want to print the `? ThenExpr :` even if the CondExpr and ElseExpr are null (as in that case they'll be printed by the SequenceExpr). Also `if (auto condExpr = expr->getCondExpr(); condExpr != nullptr)` is redundant, you just need `if (auto condExpr = expr->getCondExpr())` to check for null.